server/src/frameworks/shared/buildBasicCompilation.ts

```diff
@@ -45,17 +48,22 @@ export async function buildBasicCompilation(
   }
 
   // Build solc input
-  const filePathsToCompile = _.map(dependencyDetails, "absolutePath");
   const sources: { [uri: string]: { content: string } } = {};
-  for (const filePath of filePathsToCompile) {
+  const remappings: string[] = [];
+
+  for (const { sourceName, absolutePath } of dependencyDetails) {
```

Comment:
Can you walk me through what triggered this switch to absolute paths?

Reply:
We always used absolute paths for building basic compilation (projectless, foundry). In the particular case of foundry, after building the basic compilation we transform the source keys to root-relative imports. 

Some extra logic was added to `getDependenciesAndPragmas`, and used by `buildBasicCompilation`, in which we append to remappings the resolved source names. E.g. if the framework adapter resolved `@source/name.sol` to `/absolute/path.sol`, then on solc input's sources the key would be `/absolute/path.sol` and there'll be a remapping entry `'@source/name.sol' => '/absolute/path.sol'`. This leverages the adapter's ability to resolve paths, for helping building the solc input. Notice that in the ape adapter, no further remappings parsing is needed for getting a correct compilation. All is leveraged from `buildBasicCompilation`.